pkg/controller/mysqlcluster/mysqlcluster_controller.go

```diff
@@ -146,30 +147,29 @@ func (r *ReconcileMysqlCluster) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 	log.Info("syncing cluster", "cluster", request.NamespacedName.String())
+	wCluster := mysqlcluster.New(cluster)
```

Comment:
We can define `cluster := mysqlcluster.New(&mysqlv1alpha1.MysqlCluster{})` and use `cluster.Unwrap()` when appropriate.